src/Persistence_matrix/concept/PersistenceMatrixColumn.h

```diff
@@ -0,0 +1,469 @@
+                          Column_settings* colSettings = nullptr);
+  /**
+   * @brief Copy constructor with row access.
+   * If @p operators or @p cellConstructor is not a null pointer, its value is kept
+   * instead of the one in the copied column.
+   * 
+   * @tparam Row_container_type  Either std::map if @ref PersistenceMatrixOptions::has_removable_rows is true or
+   * std::vector<Row_type>.
+   * @param column Column to copy.
+   * @param columnIndex @ref MatIdx column index of the new column once copied.
+   * @param rowContainer Pointer to the row container that will be forwarded to @ref Row_access.
+   * @param colSettings Pointer to a setting structure or `nullptr`. The structure should contain all the necessary
+   * classes specific to the column type, such as custom allocators. The specificities are this way hidden behind
+   * a commun interface for all column types. If @p colSettings is not specified or is equal to `nullptr`, the structure
+   * stored in @p column is used instead.
+   */
+  template <class Row_container_type>
+  PersistenceMatrixColumn(const PersistenceMatrixColumn& column, 
+                          index columnIndex, 
+                          Row_container_type* rowContainer,
+                          Column_settings* colSettings = nullptr);
+  /**
+   * @brief Move constructor.
+   * 
+   * @param column Column to move.
+   */
+  PersistenceMatrixColumn(PersistenceMatrixColumn&& column) noexcept;
+  /**
+   * @brief Destructor.
+   */
+  ~PersistenceMatrixColumn();
+
+  /**
+   * @brief Returns the values of the column, zero values included.
+   * 
+   * @param columnLength Number of rows to be returned. If -1, the number of rows is fixed at the biggest 
+   * row index with non zero value. Default value: -1.
+   * @return Vector of @ref Field_element_type. At element \f$ i \f$ of the vector will be stored the value
+   * at row \f$ i \f$ of the column.
+   */
+  std::vector<Field_element_type> get_content(int columnLength = -1) const;
+  /**
+   * @brief Indicates if the cell at given row index has value zero.
+   * 
+   * @param rowIndex Row index to look at.
+   * @return true If the cell has value zero.
+   * @return false Otherwise.
+   */
+  bool is_non_zero(id_index rowIndex) const;
+  /**
+   * @brief Indicates if the column is empty or has only zero values.
+   * 
+   * @return true If the column is empty or only has zero values.
+   * @return false Otherwise.
+   */
+  bool is_empty();
+  /**
+   * @brief Returns the size of the underlying container.
+   *
+   * @warning Depending of the column type, the container does not have to contain only the non-zero cells.
+   * Even if for most of the types, the size of the container will correspond to the number of non-zero values
+   * in the column, it is not always the case. See description of the actual Column class for more details.
+   * 
+   * @return Size of the underlying container.
+   */
+  std::size_t size() const;
```

Comment:
What is it useful for? And more importantly, since this is a concept, if I implement my own column, what property does the number I return need to have?

Reply:
I admit that I only use it for testing and debugging. I should mention this in the doc or remove it from the concept (but the tests won't work without, so it is perhaps better to keep it or change the tests?).

Reply:
How do you use it in the tests, if it doesn't return the same thing for all column types?

Reply:
As long as no additions or removals are made, the size is the same for all the current columns types. And for the cases where there are some, I just make exception for some column types. Which is not ideal, but it works well enough for now.